libs/blocks/merch/autoblock.js

```diff
@@ -35,9 +35,62 @@ export async function decorateCardCtasWithA11y(element, self) {
   });
 }
 
+export function handleCustomAnalyticsEvent(eventName, element) {
+  let daaLhValue = '';
+  let daaLhElement = element.closest('[daa-lh]');
+  while (daaLhElement) {
+    if (daaLhValue) {
+      daaLhValue = `|${daaLhValue}`;
+    }
+    const daaLhAttrValue = daaLhElement.getAttribute('daa-lh');
+    daaLhValue = `${daaLhAttrValue}${daaLhValue}`;
+    daaLhElement = daaLhElement.parentElement.closest('[daa-lh]');
+  }
+  if (daaLhValue) {
+    console.log('daaLhValue: ', `${eventName}|${daaLhValue}`);
+    // eslint-disable-next-line no-underscore-dangle
+    window._satellite?.track('event', {
+      xdm: {},
+      data: { web: { webInteraction: { name: `${eventName}|${daaLhValue}` } } },
+    });
+  }
+}
+
+export function getTabNestedLh(el) {
+  const tabPanel = el.closest('.tabpanel');
+  const tabDaaLh = tabPanel?.getAttribute('data-nested-lh');
+  return tabDaaLh ? `${tabDaaLh}--tab` : '';
+}
+
+function enableCardAnalytics(el, self) {
+  const tabPanel = el.closest('.tabpanel');
+  const tabDaaLh = tabPanel?.getAttribute('data-nested-lh') ? `${tabPanel.getAttribute('data-nested-lh')}--tab` : '';
+
+  const cards = self ? [el] : el.querySelectorAll('merch-card');
+  cards.forEach(async (card) => {
+    await card.checkReady();
+    card.setAttribute('name', card.getAttribute('daa-lh'));
+    if (tabDaaLh) {
+      card.setAttribute('data-block', '');
+      card.setAttribute('daa-lh', tabDaaLh);
+    }
```

Comment:
i would avoid copying tabDaaLh over to _all_ cards, if they belong to a collection that can hold that info once.

Reply:
i think it would be better in the end with less (useless) dom write